src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size = 0;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            elements = new Object[capacity];
+        } else if (capacity == 0) {
+            elements = new Object[DEFAULT_CAPACITY];
+        } else {
+            throw new IllegalArgumentException("Invalid capacity");
+        }
+    }
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            elements = grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        if (size == this.elements.length) {
+            elements = grow();
+        }
+        if (index == elements.length) {
+            add(value);
+        } else {
+            Object[] newData = elements.clone();
+            System.arraycopy(newData, 0, elements, 0, index);
+            elements[index] = value;
+            System.arraycopy(newData, index, elements, index + 1, newData.length - index - 1);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] newData = new Object[list.size()];
+        for (int i = 0; i < newData.length; i++) {
+            newData[i] = list.get(i);
+        }
+        elements = Arrays.copyOf(elements, (size + newData.length));
+        System.arraycopy(newData, 0, elements, size, newData.length);
+        size += newData.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return (T) elements[index];
```

Comment:
Better to move type-cast to creation phase
```suggestion
        return elements[index];
```